### PR TITLE
Add documentation for databricks_pipeline & make test-preview runnable

### LINKED ISSUE #606: make test-preview should be passing with every release
Currently preview acceptance tests (`make test-preview`) are not running with every release, though it's high time they should. Create all-enabled workspace internally and include it in the release procedure.

### PR DESCRIPTION
**Pull request by nfx**
This pr fixes #606

### CODE REVIEW

**Comment by nfx** [comment]
preview
---
 * [ ] TestPreviewAccDashboard (862.77s)
 * [ ] TestPreviewAccQuery (185.40s)
 * [ ] TestPreviewAccSQLEndpoint (335.28s)
 * [ ] TestPreviewAccSQLEndpoints (1200.62s)
 * [x] compute/acceptance.TestPreviewAccPipelineResource_CreatePipeline (11.14s)

File: docs/resources/pipeline.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+```
+
+## Argument Reference
+
+The following arguments are required:
+
+* `name` - A user-friendly name for this pipeline. The name can be used to identify pipeline jobs in the UI.
+* `storage` - A location on DBFS or cloud storage where output data and metadata required for pipeline execution are stored. By default, tables are stored in a subdirectory of this location.
+* `configuration` - An optional list of values to apply to the entire pipeline. Elements must be formatted as key:value pairs.
+* `library` block - An array of notebooks containing the pipeline code and required artifacts. Syntax resembles [library](cluster.md#library-configuration-block) configuration block with the addition of special `notebook` type of library.
```
Comment (alexott) [inline docs/resources/pipeline.md:57]: `of special` -> `of a special`

File: docs/resources/pipeline.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+* `cluster` block - An array of specifications for the [clusters](cluster.md) to run the pipeline. If this is not specified, pipelines will automatically select a default cluster configuration for the pipeline.
```
Comment (alexott) [inline docs/resources/pipeline.md:58]: it's not an array, but multiple blocks - maybe rephrase it?

File: docs/resources/pipeline.md (hunk 1 above)
Comment (alexott) [inline docs/resources/pipeline.md:57]: also - it's not an array, but at least one block.

**Review by alexott — COMMENTED** [review]
Code looks good, not clear why we get a panic in the Update function

**Comment by nfx** [comment]
> Code looks good, not clear why we get a panic in the Update function

Should be fixed now. It was related to failure handling on continuous pipelines.

**Review by alexott — APPROVED** [review]

**Comment by nfx** [comment]
preview
---
 * [ ] TestPreviewAccSQLEndpoints (1200.95s)
 * [x] compute/acceptance.TestPreviewAccPipelineResource_CreatePipeline (393.72s)
 * [x] sqlanalytics/acceptance.TestPreviewAccDashboard (302.75s)
 * [x] sqlanalytics/acceptance.TestPreviewAccQuery (307.88s)
 * [x] sqlanalytics/acceptance.TestPreviewAccSQLEndpoint (357.09s)
